app/serializers/subject_selector_serializer.rb

```diff
@@ -31,6 +31,10 @@ def already_seen
     !!(user_seen&.subjects_seen?(@model.id))
   end
 
+  def favorite
+    user&.collections&.where(favorite: true)&.first&.subjects&.include? @model
```

Comment:
Does this just compare the first user favourite collection for inclusion even though they will have many, i.e. multiple projects? the serializer module to filter the has many's uses a join query to determine the favorite collections that a subject belongs to. That is they type of query this needs

I think this will result in N+1's in the serializer, i.e. each model in the serializer page will request the user's first collection and test the include subjects. It would be better to pass in the predefined set of fav subject ids to compare the model.id against. 

However I am not sure exactly how this will work but one solution could be in the controller behind the feature flag, run subject selection, get the subject ids, run the join query with the subject_ids then you could reduce the result to have a { subject_id: fav_or_not_bool } or [ fav_subject_ids_array ] that can be tested in the serializer here. This would ensure the query is only run once and not every time, thus saving some computation cost on a critical end point that we really need to be as fast as possible.